doc/guidelines/3rdparty.rst

```diff
@@ -0,0 +1,43 @@
+Third Party Libraries
+=====================
+
+Third party libraries under ``libs/3rdparty`` are maintained by
```

Comment:
It is recommended if possible.
Some 3rdparty code may not be available via (public) git repo or we want just a few files.

-->  Explain also the fallback.